strikt-core/src/main/kotlin/strikt/internal/AssertionNode.kt

```diff
@@ -48,8 +49,8 @@ internal class AssertionSubject<S>(
   }
 
   private val _children = mutableListOf<AssertionNode<*>>()
-  override val children: List<AssertionNode<*>>
-    get() = _children
+  override val children: List<AssertionNode<*>> =
+    Collections.unmodifiableList(_children)
```

Comment:
I don’t understand the purpose of this change.

Reply:
it makes it even more immutable :) in the original code you could just cast to a mutable list type.  maybe its overkill. actually i wanted to put this in a separate commit because i was not sure.